### PR TITLE
sundials-codes-addons

### PR DESCRIPTION
**Pull request by balos1**
Todo:
- [x] add docs 
- [x] Discuss this on Tuesday.

### CODE REVIEW

**Comment by balos1** [comment]
I added "captions" to the superbuild toctree (and separated the toctree into multiple) so that different parts of the docs (user guide, developer guide) are more clearly delineated in the sidebar on RTD.

File: doc/requirements.txt
Comment (gardner48) [inline doc/requirements.txt:6]: The `graphviz` python package isn't necessary to build the docs. There needs to be a [Graphviz installation](https://graphviz.org/download/) on the system (this is also required by the python package). This requirement is noted in `doc/README.md`

File: doc/README.md
Comment (gardner48) [inline doc/README.md:9]: Graphviz needs to be installed separately (see [this comment](https://github.com/LLNL/sundials/pull/444/files#r1545523642))

File: external/CMakeLists.txt
Comment (gardner48) [inline external/CMakeLists.txt:29]: Should we comment this block out or tell users to comment it out unless they want to build the example addon?

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:1]: The top-level `CONTRIBUTING.md` should be similarly updated

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:96]: This link in no longer vaild

File: external/CMakeLists.txt
Comment (balos1) [reply external/CMakeLists.txt:29]: I think we can tell them to comment it out or guard it with another variable. It would be nice to have it uncommented so we can test the external addon functionality in our CI/nightly.

File: doc/shared/sundials/Install.rst
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2200]: To make the original ``:ref:`here <Style>` `` reference work, add `'sphinx.ext.intersphinx'` to the `extensions` list in `conf.py` and add the following as well
```
intersphinx_mapping = {'sundials': (f'https://sundials.readthedocs.io/en/{doc_version}',
                                    ('../../../superbuild/build/html/objects.inv', None))}
```
When a cross reference can't be found in the build Sphinx will try to resolve it using the online version (or the local superbuild as a backup). The package guides use this to make "external" references to other packages work in the combined changelog.

File: doc/install_guide/source/conf.py
Comment (gardner48) [inline doc/install_guide/source/conf.py:34]: Need to add `'sphinx.ext.intersphinx'` to the `extensions` list above

**Comment by balos1** [comment]
@gardner48 I think this can be merged now once you approve

File: CONTRIBUTING.md
Comment (gardner48) [inline CONTRIBUTING.md:73]:
```suggestion
SUNDIALS and that you are willing to have it used in distributions and
```

File: CONTRIBUTING.md
Comment (gardner48) [inline CONTRIBUTING.md:113]:
```suggestion
1. In your addon project, ensure that you have a `CMakeLists.txt` that uses the
```

File: CONTRIBUTING.md
Comment (gardner48) [inline CONTRIBUTING.md:115]:
```suggestion
   copy from, or refer to, a `CMakeLists.txt` in the SUNDIALS `src` directory.
```

File: doc/shared/sundials/Install.rst
Comment (gardner48) [inline doc/shared/sundials/Install.rst:2212]:
```suggestion
3. When building SUNDIALS, set the CMake option :cmakeop:`SUNDIALS_ENABLE_EXTERNAL_ADDONS` to `ON`
```

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:36]: For a follow-on PR, we need to update the Pull Request instructions.

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:122]:
```suggestion
If you've set ``user.name`` and ``user.email`` in your Git configuration, you can
automatically add a sign off line at the end of the commit message by using the
``-s`` option (e.g., ``git commit -s``).
```

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:141]:
```suggestion
1. In your addon project, ensure that you have a ``CMakeLists.txt`` that uses the
```

File: doc/superbuild/source/contributing/index.rst
Comment (gardner48) [inline doc/superbuild/source/contributing/index.rst:143]:
```suggestion
   copy from, or refer to, a ``CMakeLists.txt`` in the SUNDIALS ``src`` directory.
```

**Review by gardner48 — CHANGES_REQUESTED** [review]
Minor fixes, otherwise this looks good

**Review by gardner48 — APPROVED** [review]

**Review by gardner48 — CHANGES_REQUESTED** [review]
Just noticed the change logs need to be updated to note the addition of support for external addons.

**Review by gardner48 — APPROVED** [review]
Changes look good. Read the docs test is currently failing but that's because of a bug in Sphinx 7.3.0. A [patch was released](https://www.sphinx-doc.org/en/master/changes.html#release-7-3-1-released-apr-17-2024) today so once RTD updates the docs should build again.

**Pull request merged**
